src/modules/timer.rs

```diff
@@ -0,0 +1,83 @@
+use crate::config::TableExt;
+use ansi_term::Color;
+
+use super::{Context, Module};
+
+/// Outputs the time it took the last command to execute
+///
+/// Will only print if last command took more than two seconds to execute.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("timer")?;
+    let arguments = &context.arguments;
+    let elapsed = arguments
+        .value_of("elapsed_time")
+        .unwrap_or("x") // not a time
+        .parse::<u64>()
+        .ok()?;
+
+    // Question: Is this a good series of calls to use? Is there a cleaner way to set types?
+    // Question: If the input config is negative, should I warn the user?
+    let config_min = context
+        .config
+        .get_module_config("timer")
+        .and_then(|table| table.get_config("min_time"))
+        .and_then(|value| value.as_integer())
+        .unwrap_or(2) as u64;
+
+    let module_color = match elapsed {
+        time if time < config_min => return None,
+        _ => Color::Yellow.bold(),
+    };
+
+    module.set_style(module_color);
+    module.new_segment("timer", &format!("took {}", render_time(elapsed)));
+    module.get_prefix().set_value("");
+
+    Some(module)
+}
+
+/// Render the time into a nice human-readable string
+fn render_time(raw_seconds: u64) -> String {
```

Comment:
We already have [`uom`](https://crates.io/crates/uom) as a sub-dependency of `battery`.
Maybe we should use that for zero-cost conversions here?

Reply:
@matchai My impression after reading the `uom` doc is that it's great if we want to convert seconds into hours, or minutes into days, but if we have a number of seconds that we want to display as DD:HH:MM:SS, I don't see a way to do it aside from writing a custom `Formatter`, which would basically replicate what we're doing now, but in the library.

Is there something I missed while skimming the docs? I'm still not quite used to reading rust docs yet.

Reply:
Looks like you're right. 👍